validator/src/test/java/com/amazon/aoc/validators/LoadBalanceValidatorTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.aoc.validators;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.amazon.aoc.callers.HttpCaller;
+import com.amazon.aoc.exception.BaseException;
+import com.amazon.aoc.exception.ExceptionCode;
+import com.amazon.aoc.models.CloudWatchContext;
+import com.amazon.aoc.models.Context;
+import com.amazon.aoc.models.SampleAppResponse;
+import com.amazon.aoc.models.ValidationConfig;
+import com.amazon.aoc.services.XRayService;
+import com.amazonaws.services.xray.model.Trace;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import org.junit.Test;
+
+/** this class covers the tests for LoadBalanceValidator. */
+public class LoadBalanceValidatorTest {
+
+  /** test validation when it succeeds */
+  @Test
+  public void testValidationSucceeds() throws Exception {
+    ValidationConfig validationConfig = new ValidationConfig();
+    validationConfig.setCallingType("http");
+    Context context = initContext();
+
+    String mockedTrace =
+        "[{ \"segments\": [{ \"document\": '{ \"metadata\": { \"default\": {\"collector-id\": 3} }, \"subsegments\": [{ \"metadata\": { \"default\": {\"collector-id\": 3} }}, { \"metadata\": { \"default\": {\"collector-id\": 3} }}] }' }] }]";
```

Comment:
Having to manually escape JSON strings is a painful developer experience. Let's find another way to provide this mocked trace JSON. Possibly a testdata json file or some type of raw string if java supports it.

Reply:
If we could use validator with jdk17, we could use multiline strings and therefore no scaping would be needed.

I did a bit of research and other solutions include using hashmaps to build the json https://www.baeldung.com/java-org-json#1-creating-jsonarray

no good solution here :(

Reply:
Can we just read in and deserialize from a `.json` file instead?

Reply:
I saw that with a higher JDK version we would use multiline strings which would be helpful but not sure if we want to do that.  I think reading in and deserializing from a `.json` could be a good alternative but we would have to create a folder and manage those files.

Reply:
I've added what it would look like to read from a `.json` file for the input to the first testcase and it still does not look the best.  I found out that even reading in `.json` is not much better because the value of the `document` key needs to be a string and `.json` does not allow for multi-line strings as well.  This means we still have a long escape json string.  I'm not sure if there are any other alternatives to this.  What do we think?

Reply:
We already use mustache files heavily in the validator. Can we reuse and extend the mustache rendering functionality for expected testcase files?

Reply:
The only difference is that we wouldn't need to inject any data into the mustache files. Unless you find a need to inject data during rendering.

Reply:
I now switched to rendering mustache files and it works.